core/src/main/scala/org/apache/spark/h2o/converters/H2ODataFrame.scala

```diff
@@ -92,31 +94,27 @@ class H2ODataFrame[T <: water.fvec.Frame](@transient val frame: T,
         expectedTypes,
         partIndex)
 
+      /*a sequence of converters, per column*/
+      private val columnConverters = filteredTypes map converterCtx.get
+
+
+      private def readRow: InternalRow = {
+          val optionalData: Seq[Option[Any]] =
```

Comment:
I would be interested in comparing performance of this code - will create JIRA for that